tests/src/JIT/opt/ObjectStackAllocation/ObjectStackAllocationTests.cs

```diff
@@ -191,27 +213,39 @@ static int AllocateSimpleClassesAndNECompareThem()
         {
             SimpleClassA a1 = new SimpleClassA(f1, f2);
             SimpleClassA a2 = (f1 == 0) ? a1 : new SimpleClassA(f2, f1);
+            GC.Collect();
             return (a1 != a2) ? 1 : 0;
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        static int AllocateSimpleClassAndCheckType()
+        static int AllocateSimpleClassAndCheckTypeNoHelper()
+        {
+            object o = (f1 == 0) ? (object)new SimpleClassB(f1, f2) : (object)new SimpleClassA(f1, f2);
+            GC.Collect();
+            return (o is SimpleClassB) ? 0 : 1;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static int AllocateSimpleClassAndCheckTypeHelper()
         {
             object o = (f1 == 0) ? (object)new SimpleClassB(f1, f2) : (object)new SimpleClassA(f1, f2);
-            return (o is SimpleClassB) || !(o is SimpleClassA) ? 0 : 1;
+            GC.Collect();
+            return !(o is SimpleClassA) ? 0 : 1;
```

Comment:
I bet there should be a reason why you prefered this over `(o is SimpleClassA) ? 1 : 0 `

Reply:
It's the result of refactoring the previous version of the test. I didn't bother simplifying this.